### DISCUSSION TITLE
URL Actions / Overrides

### DISCUSSION THREAD

**Discussion by MKobek** [keepassxreboot/keepassxc#8491] [Ideas]
Hi everyone. 
I like to recommend a feature which I remember from the "legacy" KeePass. Maybe it is already there but I have not found it in the documentation so far. 

If we now click on the URL, the system default browser does open and is loading the URL. I would love if we can configure this a bit one way or another.

- Open the default browser in incognito mode
- Open another browser
- Open a complete different application like... Putty, mstsc.exe or something completely different.

See the following links as an inspiration. Would be amazing if we could have this and launch whatever application with parameters this way.

**Comment by droidmonkey** [comment]
You can use the `cmd://` syntax in the url field to launch whatever you like with whatever command line parameters and even field injections using the reference syntax. 

Defining an alternative browser is already a feature request we are considering.

**Reply by MKobek** [reply]
Ah cool, seems to do the trick.
I like to recommend a note on https://keepassxc.org/docs/KeePassXC_UserGuide.html for that feature which is very helpful. Thank you very much.

**Comment by BDaddyG** [comment]
Good day!

@droidmonkey I just tried this and while the correct browser (Chrome) opened, it broke the look up for the username and password, which prevented the auto fill of the fields.

The command I used in the entry URL: field:   cmd://"/usr/bin/google-chrome" "https://smartdata.mastercard.com"
(I also tried without the double quotes around the site URL).

I added the site URL to the "Additional URLs" section of "Browser Integration" hoping it would help, but it didn't.

Once I removed the "cmd://..." and went to the site, KeepassXC was able to lookup and auto fill the username and password again.

Thanks for any insights as to why this did not work.

Linux Mint 20.3
KeepassXC v2.7.4

**Reply by droidmonkey** [reply]
That's odd, certainly the additional urls works, I use them all the time.

**Reply by rna159753** [reply]
Sorry to be a bother, but is there a guide anywhere for syntax of the cmd:// with examples?

All I'm trying to do is edit the URL field on an Entry to launch a browser in Private/Incognito mode and navigate to the url indicated.

I did search Discussions and found a few threads on using CMD:// but, man, it's just not working for me. I've tried about every combination except the one that works!

I do have a suggestion, but don't know if you would want to add it. In the Browser Integration section for the Entry, add a check-box to launch a browser instance in Private mode?

THANK YOU!

**Comment by spoonerarthur** [comment]
Hello,
I am searching for "Open URL in Browser" Action like in the other Keepass Version.
![image](https://github.com/user-attachments/assets/3a373bd9-51ba-495e-ba57-cb66b671be51)

I want to choose the browser to open the URL.

Regards Arthur

**Reply by droidmonkey** [reply]
You will have to use a cmd:// url to point to the browser you want to use. We don't support alternative browsers yet.

**Comment by spoonerarthur** [comment]
OK, thank you.
That's not quite what I need.
Then I hope that the function will be integrated soon.

**Comment by M-S3vN** [comment]
I also wish I had this "URL Overwrite" feature.
But for a different reason: I often need to use Putty to create an SSH connection or open a Windows remote desktop connection. OK - there is a workaround for ssh:// - over WinSCP. But RDP does not work for me: 
cmd://%windir%\system32\mstsc.exe /v:10.10.1.9
When I click on it, nothing happens.
When I enter it in a CMD shell it works.

By the way: when I work with variables in the URL field, the field is not green but red. But it works.
scp://{username}@192.168.100.20

**Comment by xXPerditorXx** [comment]
I too would love to have this features. Ist there any plan if adding this feature?

**Comment by Ciantic** [comment]
I think this feature is still needed. 

Putting `cmd://` in the URL makes the database difficult to use cross platform. Like from Linux and Windows. I'd prefer my URLs to be in a sane format, like:

ssh://example.com

Then with override I define how ssh:// URLs are executed. That is how KeePass works. Commands to execute are wildly different depending on environment.

**Reply by droidmonkey** [reply]
You can absolutely use ssh:// but then you need to define a URI handler in your OS for that schema.

**Reply by Ciantic** [reply]
@droidmonkey URI handler with password, username, and port injection?

All my entries are like this:

ssh://example.com

It doesn't include username and password in URI?

I don't find URI handlers the right approach, as I want to start different programs with KeePassXC than xdg-open for instance.

**Reply by droidmonkey** [reply]
https://github.com/leonseng/windows-ssh-url-handler

As an example. We are considering this capability built into KeePassXC.

**Reply by CC3301** [reply]
@droidmonkey is there any update on the state of this?

As far as my research goes, there is no way to achieve the desired behavior with using URI handlers. If you want i can provide more detail about my findings, in short, there are a few common special characters (`@` and `%`), which when present in the URI, make xdg-open fall back to the default webbrowser.

**Reply by droidmonkey** [reply]
Haven't worked on it and don't have the time to
